hippod/report_generator.py

```diff
@@ -132,15 +160,18 @@ def sanitize_description(description_path):
                         output_text.write(line)
 
 
-        def check_image_reference(description_path, attach_path):
+        def check_image_reference(description_path, attach_path, data_type):
+            data_type = data_type.replace(".","")
             reference_avaible = False
             head, tail = os.path.split(attach_path)
             with open(description_path, 'r') as input_text:
                 in_descr = input_text.readlines()
             with open(description_path, 'w') as output_text:
                 for line in in_descr:
-                    match = re.search(r'(\()(.*[.]png)', line)       # jpeg,...  # enough to assume there's a reference?
-                    p = re.compile(r'\(.*[.]png')
+                    regex = r'(\()(.*[.]' + '{})'.format(data_type)
```

Comment:
no, this is really inefficient! creating a regex is CPU intensive, doing it in every for loop without any modification makes no sense. Do it before the for loop!